src/output/plugins/SndioOutputPlugin.hxx

```diff
@@ -20,6 +20,38 @@
 #ifndef MPD_SNDIO_OUTPUT_PLUGIN_HXX
 #define MPD_SNDIO_OUTPUT_PLUGIN_HXX
 
+#include "../OutputAPI.hxx"
+#include "mixer/Listener.hxx"
+
 extern const struct AudioOutputPlugin sndio_output_plugin;
 
+class PulseMixer;
+
+class SndioOutput final : AudioOutput {
+	friend class SndioMixer;
```

Comment:
No `friend`, please, unless ultimately necessary! `friend` can access any private, and is obscure. Provide an API instead of friending other classes.